### ISSUE TITLE
Combine module

### ISSUE THREAD

**Issue by andregeges** [pingpong-labs/modules#135]
Hi, it is possible to combine different module from different project ?
how to achieve that ? am i can do it with only copy and paste entire folder of module ?

case: 
i and my friend create different module from separate project, and we want to combine our module together when we finished

**Comment by gravitano** [comment]
Of course you can. You can copy and paste the module to other project you want manually. Or if you want to open sourcing the module, you publish the module to packagist and you can install that module using composer.

**Comment by andregeges** [comment]
How to publish to packagist ?

**Comment by gravitano** [comment]
You can follow this [instruction](http://sky.pingpong-labs.com/docs/2.0/modules#publishing-modules). Or you can follow publishing `package` instruction in [packagist website](https://packagist.org/).

**Comment by andregeges** [comment]
Hmm, my modules is not open source, just private use only.. then how i can achieve that ?

**Comment by gravitano** [comment]
You can store your module to `bitbucket` maybe and make it a private repository. Now, you can install your module via composer using VCS repository -> https://getcomposer.org/doc/05-repositories.md#vcs

**Comment by andregeges** [comment]
Thanks i'll try it

**Comment by andregeges** [comment]
This is my composer.json
![screen shot 2015-04-24 at 5 07 08 pm](https://cloud.githubusercontent.com/assets/5994555/7316869/e6fba778-eaa4-11e4-9987-e27e00abf361.png)

and when i composer update i got this error
![screen shot 2015-04-24 at 5 06 54 pm](https://cloud.githubusercontent.com/assets/5994555/7316848/bcf8e15c-eaa4-11e4-9278-2bd2421e9802.png)

**Comment by gravitano** [comment]
That is error in your `composer.json` syntax. Looks like you are missing comman `,` after `[`.
```
"repositories": [

], <--
```

**Comment by andregeges** [comment]
Yes, then i got new problem

Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - The requested package git@bitbucket.org:andregeges/core.git could not be found, it looks like its name is invalid, "@:" is not allowed in package names.

Potential causes:
 - A typo in the package name
 - The package is not available in a stable-enough version according to your minimum-stability setting
   see <https://groups.google.com/d/topic/composer-dev/_g3ASeIFlrc/discussion> for more details.

**Comment by gravitano** [comment]
```
- The requested package git@bitbucket.org:andregeges/core.git could not be found, it looks like its name is invalid, "@:" is not allowed in package names.
````

Your package name is not allowed by composer. You need to follow composer naming convention. Read this helpful documentation -> https://getcomposer.org/doc/04-schema.md#name.

**Comment by andregeges** [comment]
yes after read that i dont know where is wrong
maybe because of version??
how to add version to my repos ?
----
version#

The version of the package. In most cases this is not required and should be omitted (see below).

This must follow the format of X.Y.Z or vX.Y.Z with an optional suffix of -dev, -patch (-p), -alpha (-a), -beta (-b) or -RC. The patch, alpha, beta and RC suffixes can also be followed by a number.

Examples:

    1.0.0
    1.0.2
    1.1.0

**Comment by gravitano** [comment]
There is no problem with your version. But in your package name. Package
must separate between vendor and package name using /. Example :
vendor/package.

If you want to make versioning of your module/package. You can create tag
for the repository that you use and dont forget to push the tag to the
repository.

**Comment by andregeges** [comment]
do you mean this ? 

andregeges/core.git

vendor = andregeges
package = core

**Comment by gravitano** [comment]
Without `.git`. `andregeges/core`.

**Comment by andregeges** [comment]
still not working, same error..
can you give package example using this method ?
so i will try to add your module to my composer.json using this method.

anyway this is my private repo but i dont write any credential to get my repo. am i missing something ?

**Comment by andregeges** [comment]
- Installing andregeges/core (dev-master 56be779)
    Cloning 56be77975cb965a8bab038ddf486dbd74a6fb1a8

Writing lock file
Generating autoload files
[RuntimeException]                                                                                                                                          
  Could not scan for classes inside "/Applications/MAMP/htdocs/package2/vendor/andregeges/core/database" which does not appear to be a file nor a folder

**Comment by gravitano** [comment]
Here is an example of pingpong module and also submitted to packagist. 
https://github.com/pingpongcms/Hello
https://packagist.org/packages/pingpong-modules/hello

**Comment by andregeges** [comment]
still got some error..
maybe i need search some complete documentation and step by step tutorial..

---

Writing lock file
Generating autoload files            
                                           
  [Symfony\Component\Debug\Exception\FatalErrorException]                             
  Call to undefined method Illuminate\Foundation\Application::getCachedCompilePath()  

Script php artisan clear-compiled handling the post-update-cmd event returned with an error                 
  [RuntimeException]  
  Error Output:       
                      
update [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--lock] [--no-plugins] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [--with-dependencies] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [packages1] ... [packagesN]

**Comment by gravitano** [comment]
That error occurs because you do not delete `compiled` file. Just delete the `compiled.php` file & run `php artisan optimize` to fix that problem.

**Comment by andregeges** [comment]
Ok, it's work..

now how to call your route ? i got this error when i access this routes
http://localhost:8888/package/public/hello

i'm already do composer dump-autoload
but still got this error

---

Sorry, the page you are looking for could not be found.

1/1 NotFoundHttpException in RouteCollection.php line 145:

    in RouteCollection.php line 145
    at RouteCollection->match(object(Request)) in Router.php line 729
    at Router->findRoute(object(Request)) in Router.php line 652
    at Router->dispatchToRoute(object(Request)) in Router.php line 628
    at Router->dispatch(object(Request)) in Kernel.php line 214
    at Kernel->Illuminate\Foundation\Http\{closure}(object(Request))
    at call_user_func(object(Closure), object(Request)) in Pipeline.php line 141
    at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in VerifyCsrfToken.php line 43
    at VerifyCsrfToken->handle(object(Request), object(Closure)) in VerifyCsrfToken.php line 17
    at VerifyCsrfToken->handle(object(Request), object(Closure)) in Pipeline.php line 125
    at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in ShareErrorsFromSession.php line 55
    at ShareErrorsFromSession->handle(object(Request), object(Closure)) in Pipeline.php line 125
    at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in StartSession.php line 61
    at StartSession->handle(object(Request), object(Closure)) in Pipeline.php line 125
    at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in AddQueuedCookiesToResponse.php line 36
    at AddQueuedCookiesToResponse->handle(object(Request), object(Closure)) in Pipeline.php line 125
    at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in EncryptCookies.php line 40
    at EncryptCookies->handle(object(Request), object(Closure)) in Pipeline.php line 125
    at Pipeline->Illuminate\Pipeline\{closure}(object(Request)) in CheckForMaintenanceMode.php line 42
    at CheckForMaintenanceMode->handle(object(Request), object(Closure)) in Pipeline.php line 125
    at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
    at call_user_func(object(Closure), object(Request)) in Pipeline.php line 101
    at Pipeline->then(object(Closure)) in Kernel.php line 115
    at Kernel->sendRequestThroughRouter(object(Request)) in Kernel.php line 84
    at Kernel->handle(object(Request)) in index.php line 53

**Comment by gravitano** [comment]
Do you have `hello` route in your module ?

**Comment by andregeges** [comment]
yes im using your example here : https://github.com/pingpongcms/Hello

![screen shot 2015-04-25 at 8 10 30 pm](https://cloud.githubusercontent.com/assets/5994555/7332994/2c886032-eb87-11e4-9daa-0e5185f29585.png)

**Comment by gravitano** [comment]
Have you enabled the `scan` feature? https://github.com/pingpong-labs/modules/blob/2.0/src/config/config.php#L123-L128

**Comment by andregeges** [comment]
yes, i publish vendor, then update modules.php scan = true

and this is the result

![screen shot 2015-04-25 at 10 41 21 pm](https://cloud.githubusercontent.com/assets/5994555/7333521/6f4db6be-eb9c-11e4-89dd-34e80ba3669a.png)
![screen shot 2015-04-25 at 10 41 44 pm](https://cloud.githubusercontent.com/assets/5994555/7333522/70552466-eb9c-11e4-8573-8d9c40dae375.png)

**Comment by haiderghaleb** [comment]
Did anyone find a solution regarding this topic?
